widgetastic_manageiq/__init__.py

```diff
@@ -2635,6 +2637,127 @@ def fill(self, items):
+        Widget.__init__(self, parent, logger=logger)
+        if id:
+            self.locator = self.BASE_LOCATOR.format(quote(id))
+        elif locator:
+            self.locator = locator
+        else:
+            raise TypeError("You need to specify a locator for ReactSelect")
+
+        self.can_hide_on_select = can_hide_on_select
+
+    @property
+    def is_open(self):
+        try:
+            return self.browser.is_displayed('//div[@class="css-1paxw40-menu"]')
+        except StaleElementReferenceException:
+            self.logger.warning(
+                "Got a StaleElementReferenceException in .is_open, but ignoring. Returned False."
+            )
+            return False
+
+    def open(self):
+        if not self.is_open:
+            self.click()
+            self.logger.debug("opened")
+
+    def close(self):
+        try:
+            if self.is_open:
+                self.click()
+                self.logger.debug("closed")
+        except NoSuchElementException:
+            if self.can_hide_on_select:
+                self.logger.info("While closing %r it disappeared, but ignoring.", self)
+            else:
+                raise
+
+    def select_by_visible_text(self, *items):
+        """Selects items in the select.
+
+        Args:
+            *items: Items to be selected. If the select does not support multiple selections and you
+                pass more than one item, it will raise an exception. If you want to select using
+                partial match, use the :py:class:`BootstrapSelect.partial` to wrap the value.
+        """
+        if len(items) > 1:
+            raise ValueError(
+                "The ReactSelect {} does not allow multiple selections".format(self.locator)
+            )
+        self.open()
+        for item in items:
+            item = item.split(" *")[0]
+            self.logger.info("selecting by visible text: %r", item)
+            try:
+                self.browser.click(
+                    self.BY_VISIBLE_TEXT.format(quote(item)), parent=self, force_scroll=True
+                )
+            except NoSuchElementException:
+                try:
+                    # Added this as for some views(some tags pages) dropdown is separated from
+                    # button and doesn't have exact id or name
+                    self.browser.click(self.BY_VISIBLE_TEXT.format(quote(item)), force_scroll=True)
+                except NoSuchElementException:
+                    raise SelectItemNotFound(
+                        widget=self, item=item, options=[opt.text for opt in self.all_options]
+                    )
+        self.close()
+
+    @property
+    def all_selected_options(self):
+        return [
+            self.browser.text(e)
+            for e in self.browser.elements('//ul[@class="tag-category list-inline"]', parent=self)
+        ]
+
+    @property
+    def all_options(self):
```

Comment:
Some additional parsing will need to be done with the output of the command to distinguish between the `tag_catagory` and the `tag_name`. Right now it returns a big dictionary of that contains all the metadata about each `tag_category` and each `tag_name`.

Reply:
I know... but I don't see it's used anywhere except error message, so I thought it's fine to see all the possible options. I will try to rewrite it though

Reply:
right, working on it

Reply:
went another way and just open a dropdown :)

Reply:
fixed